File: src/main/java/com/pepebot/dto/TradeResponse.java
package com.pepebot.dto;

import java.math.BigDecimal;
import java.time.Instant;

public record TradeResponse(
		String tradeId,
		String orderId,
		String symbol,
		Side side,
		OrderType type,
		BigDecimal volume,
		BigDecimal price,
		String status,
		String message,
		Instant executedAt) {
}

File: src/main/kotlin/com/pepebot/dto/TradeResponse.kt
package com.pepebot.dto

import java.math.BigDecimal
import java.time.Instant

data class TradeResponse(
    val tradeId: String,
    val orderId: String,
    val symbol: String,
    val side: Side,
    val type: OrderType,
    val volume: BigDecimal,
    val price: BigDecimal?,
    val status: String,
    val message: String,
    val executedAt: Instant
)
